pandas/tests/indexing/test_coercion.py

```diff
@@ -29,11 +29,21 @@ def has_test(combo):
             klass in x.name and dtype in x.name and method in x.name for x in cls_funcs
         )
 
-    for combo in combos:
-        if not has_test(combo):
-            raise AssertionError(f"test method is not defined: {cls.__name__}, {combo}")
+    opts = request.config.option
```

Comment:
why not just parameterize on this?

Reply:
huh?

this is checking if pytest is invoked with `--lf` or `-k foo`

Reply:
sure, we don't do this anywhere else, so why is this actually needed here?

Reply:
#23930 this will produce unhelpful errors when running only a subset of tests in this file

Reply:
i saw the issue, my question is that this is either a very specific fix, in which case it *might* be ok, or something that actually is generally a problem. I would suggest that we could refactor the tests to simply avoid having to do this, rather than trying to address this very specifically.

Reply:
big-picture, i think check_comprehensiveness is a good idea and we should eventually hope to use something like it in more contexts.

even if we refactored the tests in this file (by no means a small feat), if we want to have a check for comprehensiveness it would still look something like this function

Reply:
ok fair, just don't want this to be specific here, e.g. it can easily get lost